js/slides/slidesModel.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * The base class for Slides.
+ */
+class BaseSlide {
+    constructor(label = DEFAULT_LABEL) {
+        if (!BaseSlide._id) { BaseSlide._id = 0; }
+        BaseSlide._id++;
+        this._uid = `${this.constructor.name}-${BaseSlide._id}`;
+        this._label = label;
+    }
+    /**
+     * The Id number of the slide instances created.
+     */
+    static get id() {
+        return BaseSlide._id;
+    }
+    /**
+     * Returns the unique identifier for the slide instance
+     */
+    get uid() {
+        return this._uid;
+    }
+    /**
+     * Returns the unique idenfing ID of the slide instance.
+     */
+    toString() {
+        return this._uid;
+    }
+}
+
+/**
+ * Class to hold the model of the Test Case. It holds the necessary parts of a test case
+ * including, but not limited to: the question, a series of steps, test operations, and
+ * the user answer.
+ */
+class TestSlide extends BaseSlide {
+    constructor(label) {
+        super(label);
+        /**
+         * The question or text for the slide.
+         */
+        this.question = "";
+        /**
+         * [Optional] Options for the slide. If it's a test, then it would be possible options for the 
+         * question's answer.
+         */
+        this.options = [];
```

Comment:
This may not need to be necessary. However, keep for now incase it becomes useful when we start to insert tests.